src/core/collector/mod.rs

```diff
@@ -203,6 +212,24 @@ impl Collector {
 		})
 	}
 
+	pub async fn map_subxt_event(&mut self, event: &SubxtEvent) -> color_eyre::Result<Vec<SubxtEvent>> {
```

Comment:
Neat: `map` usually means `Iter`-like semantics, so probably it is better to call it `transform_subxt_event`. Do we really need `self` here, especially as `mut`?

Reply:
I need the endpoint over self. `mut` is useless, true